orbstation/code/antagonists/contract_killer/contract_killer.dm

```diff
@@ -0,0 +1,87 @@
+/// A minor antagonist with the goal to kill a single crew member. Gets a few toys to do so.
+/datum/antagonist/contract_killer
+	name = "Contract Killer"
+	show_in_antagpanel = TRUE
+	antagpanel_category = "Other"
+	job_rank = ROLE_CONTRACT_KILLER
+	antag_hud_name = "contractkiller"
+	ui_name = "AntagInfoMinor"
+	show_name_in_check_antagonists = TRUE
+	roundend_category = "contract killers"
+	count_against_dynamic_roll_chance = FALSE
+	silent = FALSE
+	suicide_cry = "FOR THE PAY!! Wait-"
+	preview_outfit = /datum/outfit/contract_killer
+
+/datum/antagonist/contract_killer/on_gain()
+	forge_objectives()
+	return ..()
+
+/datum/antagonist/contract_killer/greet()
+	owner.current.playsound_local(get_turf(owner.current), 'orbstation/sound/antag_alerts/contractkilleralert.ogg', 100, FALSE, pressure_affected = FALSE, use_reverb = FALSE)
+	owner.announce_objectives()
+
+/datum/antagonist/contract_killer/forge_objectives()
+	var/datum/objective/assassinate/kill_objective = new
+	kill_objective.owner = owner
+	var/datum/mind/target_mind = kill_objective.find_target()
+
+	// The killer must first follow their target to gather intel in order to prepare and unlock their gear.
+	var/datum/objective/gather_intel/intel = new
+	intel.owner = owner
+	intel.target = target_mind
+	intel.start_ticking()
+	objectives += intel
+
+	objectives += kill_objective // Second objective is the kill itself.
+
+	// Finally, generate a "fluff" objective that gives extra instructions.
+	var/list/potential_fluff_objs = list()
+
+	for(var/datum/objective/contract_killer/killer_obj as anything in subtypesof(/datum/objective/contract_killer))
+		potential_fluff_objs[killer_obj] = initial(killer_obj.weight)
+
+	var/chosen_obj = pick_weight(potential_fluff_objs)
+	var/datum/objective/contract_killer/fluff_obj = new chosen_obj
+	fluff_obj.owner = owner
+	if(istype(fluff_obj, /datum/objective/contract_killer/targeted))
+		fluff_obj.find_target(blacklist = list(target_mind))
+	if(istype(fluff_obj, /datum/objective/contract_killer/newscaster))
+		fluff_obj.give_summon_action()
```

Comment:
Could you delegate this to a proc in `/datum/objective/contract_killer`?
So instead of doing typechecks, instead you call `fluff_obj.post_created(target_mind)` which is extended by these two subtypes.